app/src/main/java/com/openclassrooms/realestatemanager/ui/EstateSheetFragmentViewModel.kt

```diff
@@ -0,0 +1,39 @@
+package com.openclassrooms.realestatemanager.ui
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModel
+import com.openclassrooms.realestatemanager.model.Estate
+import com.openclassrooms.realestatemanager.repository.EstateRepository
+
+class EstateSheetFragmentViewModel(private val estateRepository: EstateRepository): ViewModel() {
+
+    private var estateListObserver: Observer<List<Estate>>
+
+    init {
+        estateListObserver = Observer { estateList ->
+            if (estateRepository.getSelectedEstateLiveData().value == null && estateList != null && estateList.isNotEmpty()) {
+                estateRepository.setSelectedEstateLiveData(estateList[0])
+            }
+        }
+        estateRepository.getEstateListLiveData().observeForever(estateListObserver)
+    }
+
+    override fun onCleared() {
+        estateRepository.getEstateListLiveData().removeObserver(estateListObserver)
+        super.onCleared()
+    }
+
+    fun getSelectedEstateLiveData(): LiveData<Estate> = estateRepository.getSelectedEstateLiveData()
+
+    fun getLocationString(estate: Estate?): String {
+        if (estate != null) {
+            if (estate.additionalAddress == null) {
+                return "{$estate.houseNumber} ${estate.street}\n${estate.city}\n${estate.zipCode}\n${estate.country}"
+            }
+            return "${estate.houseNumber} ${estate.street}\n${estate.additionalAddress}\n${estate.city}\n${estate.zipCode}\n${estate.country}"
+        }
+        return ""
+    }
```

Comment:
C'est un bon candidat pour introduire les fonctions et propriétés d'extension en Kotlin, et "l'élevation du return" : 

```suggestion
    val Estate.locationString : String get() = if (estate.additionalAddress == null) {
        "{$estate.houseNumber} ${estate.street}\n${estate.city}\n${estate.zipCode}\n${estate.country}"
    } else {
        "${estate.houseNumber} ${estate.street}\n${estate.additionalAddress}\n${estate.city}\n${estate.zipCode}\n${estate.country}"
    }
```

Ca te permet d'accéder en appelant simplement : 

```kotlin
val estate : Estate? = ...
val location = estate?.locationString ?: ""
```

Reply:
Comment tu lui passe le paramétre estate ?

Reply:
Je valide la PR pour avancer, mais on parlera de se point mercredi et je modifirai après.

Reply:
OK :) en attendant tu peux aussi consulter la doc Kotlin : https://kotlinlang.org/docs/extensions.html#extension-properties